Add breadcrumb functionality for desktop sizes

roperev opened the pull request:
Hidden on mobile view sizes.

Fix #103 

Test URLs:
- Before: https://main--idfc--aemsites.aem.page/credit-card/rupay-credit-card
- After: https://103-breadcrumbs--idfc--aemsites.aem.page/credit-card/rupay-credit-card

Additional Test URLs:
- Before: https://main--idfc--aemsites.aem.page/credit-card/metal-credit-card/mayura
- After: https://103-breadcrumbs--idfc--aemsites.aem.page/credit-card/metal-credit-card/mayura

<img width="751" height="563" alt="image" src="https://github.com/user-attachments/assets/1e652441-fbb8-4752-b69d-5e66fe98577c" />


<img width="748" height="478" alt="image" src="https://github.com/user-attachments/assets/8e6f2a69-46f3-42e9-a75e-94f55bf89132" />

Commit:
UE data structure for breadcrumbs toggle, initial boilerplate code integration

Commit:
Merge remote-tracking branch 'origin/main' into 103-breadcrumbs

Commit:
Add breadcrumb navigation implementation:

- Updated JavaScript functions in `scripts/scripts.js` to build and load breadcrumbs dynamically based on navigation structure.
- Added CSS styles for breadcrumbs in `styles/lazy-styles.css`
- Breadcrumbs are positioned as the first element in `<main>` inside the first `<blockname>-section` --> `<blockname>-wrapper` div.
- Included breadcrumb-details.md to document.

helms-charity commented on blocks/header/breadcrumb-details.md:
```diff
@@ -0,0 +1,257 @@
+# Breadcrumbs Implementation - Updated
```
what is all of this -- do we really need it? You can't speak to design decisions, only the customer can. And do we really need all the testing info for breadcrumbs? I would just add the bit about the "set breadcumbs=true if yuo want them" in the Library document and get rid of all this.

roperev replied:
Great point, this is unnecessary. I'll remove it.

helms-charity commented on scripts/scripts.js:
```diff
@@ -829,6 +829,177 @@ async function loadCategoryNav(main) {
+          .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
+          .join(' ');
+        // Pluralize "card" to "cards" for better grammar in breadcrumbs
+        title = title.replace(/\bCard\b/g, 'Cards');
```
title.replace is OK for now. I have a feeling they will ask to be able to author a "breadcrumb title" on a page-by-page basis later.

roperev replied:
Would we prefer to add a "Breadcrumb title" field to UE right now and use normal og:title as the fallback instead?

helms-charity replied:
Might as well, thank you.

helms-charity reviewed (commented):
Do you want to add LD+JSON schema with this PR? You can make a new ticket for it if you 'd rather.
<img width="639" height="674" alt="image" src="https://github.com/user-attachments/assets/b39e82aa-1728-4476-a69a-dd316357c3f6" />

It should be inline with the breadcrumbs instead of the head though. See my old project for inspiration. https://github.com/aemsites/888de/blob/954ec3ae06ab85327f64f2800efb5791d3e04384/scripts/scripts.js#L318-L336

Commit:
Remove breadcrumb-details.md and refactor header styles and scripts:

- Remove `breadcrumb-details.md`
- Updated `header.css` to remove breadcrumb-related styles
- Refactored `scripts.js` to enhance breadcrumb functionality, including the addition of a customizable breadcrumbs title
- Adjusted JSON model to include a new field for breadcrumbs title

roperev commented:
> Do you want to add LD+JSON schema with this PR? You can make a new ticket for it if you 'd rather. <img alt="image" width="639" height="674" src="https://private-user-images.githubusercontent.com/3883795/521089025-b39e82aa-1728-4476-a69a-dd316357c3f6.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQ3MjE4NDEsIm5iZiI6MTc2NDcyMTU0MSwicGF0aCI6Ii8zODgzNzk1LzUyMTA4OTAyNS1iMzllODJhYS0xNzI4LTQ0NzYtYTY5YS1kZDMxNjM1N2MzZjYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTIwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEyMDNUMDAyNTQxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjI0Mzg5NzA4ZWVjZDlhNGQwMjRlY2JkMDIyMzU1YTRmZDJiYzg1ZmJkOTk5NTVkZTNiNGQ2M2UxOTNlODBmZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.D5ExwT8EJ-_UPy83jW-NS4anVfG74nsBBaHNoj1MN9c">
> 
> It should be inline with the breadcrumbs instead of the head though. See my old project for inspiration. https://github.com/aemsites/888de/blob/954ec3ae06ab85327f64f2800efb5791d3e04384/scripts/scripts.js#L318-L336

@helms-charity Should we add this schema addition to ticket #76? I imagine the process would be similar for both. Or if you want it done as part of this issue ticket, I'll add it for this PR instead.

Commit:
Suppress descending specificity in `header.css`. Removed outdated media queries. Adjusted z-index for breadcrumbs.

roperev commented:
Final commit fixed all linting issues with `header.css` (~55 issues) except for descending specificity which has now been suppressed (8 warnings) for this file as the human readable order is preferred in this case.

helms-charity commented on component-models.json:
```diff
@@ -2,6 +2,18 @@
+        "description": "Show the breadcrumbs trail on the page (default: off)"
+      },
+      {
+        "component": "text",
```
You can add a "condition" to only show the breadcrumbsTitle if the show breadcrumbs is true. See https://github.com/aemsites/extweb-academy/blob/main/blocks/video/_video.json#L56-L67 as an example.

roperev replied:
Great point, I should have thought of that. I've added it now.

Commit:
Hide breadcrumbsTitle if Show Breadcrumbs is toggled off in UE

Commit:
Clean up UE descriptions

Commit:
fix case mismatch for breadcrumbTitle in js
